classes/ObjectModel.php

```diff
@@ -406,12 +406,16 @@ protected function formatFields($type, $id_lang = null)
             // E.g. if only lang fields are filtered, ignore fields without lang => true
             if (($type == self::FORMAT_LANG && empty($data['lang']))
                 || ($type == self::FORMAT_SHOP && empty($data['shop']))
-                || ($type == self::FORMAT_COMMON && ((!empty($data['shop']) && $data['shop'] != 'both') || !empty($data['lang'])))) {
+                || ($type == self::FORMAT_COMMON && ((!empty($data['shop']) && $data['shop'] !== 'both') || !empty($data['lang'])))) {
```

Comment:
Isn't this related to this bug? https://github.com/PrestaShop/PrestaShop/issues/25346#issuecomment-892599529

Reply:
Mmm, yes and no The part I fixed is only about partial update

I noticed the same problem mentioned in the other issue, regardless of how you defined the multishop field I had to have it in both the origin table and the multishop table, so I guess there is still a bug about this Should it be fixed in the same PR?

Reply:
I researched a bit more the other bug, so it is indeed partly related to this code but also (mainly) because of this part https://github.com/PrestaShop/PrestaShop/blob/62206a604f2557d5e067d6b8d81f68528e254d47/classes/ObjectModel.php#L296

Which merges multi shop fields with regular fields no matter what, seems like it was for retro compatibility I guess we could get rid of this in v8 so that multishop fields are no longer duplicated in their associated table? (in another PR maybe as it represents quite a big change)

Reply:
> But aside from that, if we wait for certainties to make changes I think we will never make them, the code is 7 years old, nothing was documented, all the old devs are gone I don't see how we can have certainties on what was expected at the time, we will always move forward making guesses

If we spend one or two weeks deciphering the ObjectModel behavior, carefully building test cases one by one to make sure we understand 100% the goal of each statement, we could have certainties.

However it needs a lot of time 😞